tic_tac_toe/game.py

```diff
@@ -87,18 +129,61 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    board = game['board']
+    if game['winner'] or _board_is_full(board):
+        raise InvalidMovement("Game is over.")
+        
+    elif not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+    
+    elif not _position_is_empty_in_board(position, board):
+        raise InvalidMovement('Position already taken.')
+    
+    elif get_next_turn(game) != player:
+        raise InvalidMovement('"{}" moves next'.format(game['next_turn']))
+    
+    
+    board[position[0]][position[1]] = player
+    
+    winner = _check_winning_combinations(board, player)
+    
+    if winner:
+        game['winner'] = winner
+        game['next_turn'] = None
+        raise GameOver('"{}" wins!'.format(winner))
+    
+    elif _board_is_full(board):
+        game['next_turn'] = None
+        raise GameOver('Game is tied!')
+        
+    else:
+        if game['player1'] == game['next_turn']:
+            game['next_turn'] = game['player2']
+        else:
+            game['next_turn'] = game['player1']
 
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+    
+    board = game['board']
+    board_string= """
+{}  |  {}  |  {}
+--------------
+{}  |  {}  |  {}
+--------------
+{}  |  {}  |  {}
+"""
+    
+    flattened = sum(board, [])
+    
+    return board_string.format(*flattened)
```

Comment:
You could just simplify this to `return board_string.format(*sum(board, []))` If you're worried that the sum makes it unclear what you're doing you can add a comment.